client/app.jsx

```diff
@@ -36,10 +61,20 @@ export default class App extends React.Component {
   }
 
   render() {
+    if (this.state.isAuthorizing) return null;
+    const { user, route } = this.state;
+    const { handleSignIn, handleSignOut } = this;
+    const contextValue = { user, route, handleSignIn, handleSignOut };
     return (
-      <>
-        {this.renderPage()}
-      </>
+      <AppContext.Provider value={contextValue}>
+        <>
+          {this.renderPage()}
+        </>
+      </AppContext.Provider>
     );
   }
 }
+
+App.contextType = AppContext;
+Landing.contextType = AppContext;
+Home.contextType = AppContext;
```

Comment:
It's better to import the context into the modules of these individual components so that the dependency is explicit.